### ISSUE TITLE
permissions-related example in explainer encourages bad assumptions about browser behavior

### ISSUE THREAD

**Issue by dbaron** [w3c-cg/web-nfc#482]
The permissions-related example in [the explainer](https://github.com/w3c/web-nfc/blob/gh-pages/EXPLAINER.md) currently has this code:

```js

const nfcPermissionStatus = await navigator.permissions.query({ name: "nfc" });
if (permissionStatus.state === "granted") {
  // NFC access was previously granted, so we can start NFC scanning now.
  startScanning();
} else {
  // Show a "scan" button.
  document.querySelector("button").style.display = "block";
  document.querySelector("button").onclick = event => {
    // Prompt user to allow UA to send and receive info when they tap NFC devices.
    startScanning();
  };
}
```

This makes a bunch of assumptions about browser behavior that probably shouldn't be encouraged:  in particular, that all browsers require user activation before being willing to show a permission prompt.  (This particular variation between browsers exists in the real world:  Chrome requires user activation for its WebRTC audio/video permission prompt (and then I think persists the permission); Firefox does not (but does not persist the permission by default).)

It would be better if this example were written so the code took actions based on whether activation was actually required.  Unfortunately, this can't be done today; it depends on enhancements to the Permissions API for which w3c/permissions#189 has been filed.  I'm hoping it will be doable not too far in the future, though...

(I got here from w3ctag/design-reviews#461.)

**Comment by beaufortfrancois** [comment]
You're right. This is an issue with the Permissions API hinted in the spec at https://w3c.github.io/web-nfc/#obtaining-permission.
Hopefully when https://github.com/w3c/permissions/issues/194 will be fixed, we'll be able to simply mention that user activation is required in the "request-to-use-permission" algorithm.

Note that the permissions-related example in the explainer would also work with a permission flow a la Firefox:
```js
const nfcPermissionStatus = await navigator.permissions.query({ name: "nfc" });
if (permissionStatus.state === "granted") {
  // NFC access was previously granted, so we can start NFC scanning now.
  // UPDATED: Even though it is not persisted, this state could have been granted in user session. 
  startScanning();
} else {
  // Show a "scan" button.
  document.querySelector("button").style.display = "block";
  document.querySelector("button").onclick = event => {
    // Prompt user to allow UA to send and receive info when they tap NFC devices.
    // UPDATED: User would not be prompted indeed in Firefox, but scanning would start asynchronously if success.
    startScanning();
  };
}
```
